File: Test++/Test++/LinkedList.cpp
#include "LinkedList.h"
#include <iostream>

using namespace std;

LinkedList::LinkedList()
{
	size = SIZE;
}

void LinkedList::PrintList()
{
	cout << "\nThe list contains..." << endl;
	LinkedListNode* current;
	for (current = head; current != NULL; current = current->next)
	{
		cout << current->value.movie << " " << current->value.time << endl;
	}
}

void LinkedList::AddToList(Theater item)
{
	LinkedListNode* newItem = new LinkedListNode();
	newItem->value = item;
	newItem->next = head;
	head = newItem;

	if (size == 0)
	{
		tail = newItem;
	}

	size++;
}

void LinkedList::DeleteFromList()
{
	LinkedListNode* current;
	current = head;
	head = head->next;
	size--;
}

File: Test/Test/LinkedList.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Test
{
    class LinkedList
    {
        public LinkedListNode head;
        public LinkedListNode tail;
        private int size;

        public void PrintList()
        {
            Console.WriteLine("The list contains...");
            LinkedListNode current = new LinkedListNode();
            for(current = head; current!= null; current = current.next)
            {
                current.value.printTheater();
            }
        }

        public void AddListItem(Theater item)
        {
            LinkedListNode newNode = new LinkedListNode();
            newNode.value = item;
            newNode.next = head;
            head = newNode;
            if (size == 0)
            {
                tail = newNode;
            }

            size++;
        }

        public void DeleteFromList()
        {
            LinkedListNode current = new LinkedListNode();
            current = head;
            head = head.next;
            size--;
        }

    }
}
